### PR TITLE
Increase Flannel waiting and improve log

### PR DESCRIPTION
**Pull request by camilamacedo86**
#### What this PR does / why we need it:

See that if the script fails to get the logs:

```
⚙  Addon flannel 0.21.5
namespace/kube-flannel unchanged
serviceaccount/flannel unchanged
clusterrole.rbac.authorization.k8s.io/flannel unchanged
clusterrolebinding.rbac.authorization.k8s.io/flannel unchanged
configmap/kube-flannel-cfg unchanged
secret/flannel-troubleshoot-spec configured
daemonset.apps/kube-flannel-ds unchanged
waiting for Flannel to become healthy
Error from server: Get "https://10.154.0.89:10250/containerLogs/kube-flannel/kube-flannel-ds-bznsb/kube-flannel?tailLines=10": net/http: TLS handshake timeout
An error occurred on line 146
```

It will fail with `An error occurred on line 146` instead of the bail message.
This PR improve the waiting time for 5 minutes and adds || to avoid fails with error script line instead of the bail message

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
Increase the Flannel waiting time to check that it is available and improve log in case of failures. This change are valid only for Flannel versions equals and upper than `0.21.5`. 
```

### CODE REVIEW

**Review by rrpolanco — APPROVED** [review]
LGTM. However, if we have to increase the timeout more than 5 mins. that should prompt us to investigate what's taking so long.

File: addons/flannel/0.21.5/install.sh [hunk 1]
```diff
@@ -141,9 +141,9 @@ function flannel_health_check() {
 }
 
 function flannel_ready_spinner() {
-    echo "waiting for Flannel to become healthy"
-    if ! spinner_until 180 flannel_health_check; then
-        kubectl logs -n kube-flannel -l app=flannel --all-containers --tail 10
+    echo "Waiting for Flannel to become healthy up 5 minutes"
+    if ! spinner_until 300 flannel_health_check; then
+        kubectl logs -n kube-flannel -l app=flannel --all-containers --tail 10 || true
```
Comment (laverya) [inline addons/flannel/0.21.5/install.sh:146]:
```suggestion
    echo "Waiting up 5 minutes for Flannel to become healthy"
    if ! spinner_until 300 flannel_health_check; then
        kubectl logs -n kube-flannel -l app=flannel --all-containers --tail 10 || true
```

File: addons/flannel/template/base/install.sh [hunk 2]
```diff
@@ -141,9 +141,9 @@ function flannel_health_check() {
 }
 
 function flannel_ready_spinner() {
-    echo "waiting for Flannel to become healthy"
-    if ! spinner_until 180 flannel_health_check; then
-        kubectl logs -n kube-flannel -l app=flannel --all-containers --tail 10
+    echo "Waiting for Flannel to become healthy up 5 minutes"
```
Comment (laverya) [inline addons/flannel/template/base/install.sh:144]:
```suggestion
    echo "Waiting up 5 minutes for Flannel to become healthy"
```

**Review by laverya — CHANGES_REQUESTED** [review]
grammar changes

File: addons/flannel/0.21.5/install.sh [hunk 3]
```diff
@@ -141,9 +141,9 @@ function flannel_health_check() {
 }
 
 function flannel_ready_spinner() {
-    echo "waiting for Flannel to become healthy"
-    if ! spinner_until 180 flannel_health_check; then
-        kubectl logs -n kube-flannel -l app=flannel --all-containers --tail 10
+    echo "Waiting up 5 minutes for Flannel to become healthy"
```
[the remark this answers is not in the document]
Comment (laverya) [reply addons/flannel/0.21.5/install.sh:144]: 🤦 my bad

**Review by rrpolanco — APPROVED** [review]

**Review by laverya — APPROVED** [review]

**Pull request merged**